cmd/internal/cmd/flow/flow.go

```diff
@@ -129,8 +134,17 @@ func runFlows(p *printer.Printer) error {
 	}
 
 	for i := 0; i < opts.count; i++ {
+		// used to get a random node name
 		idx := rand.Intn(len(nodesIPs))
-		t := ts.Add(time.Duration(i) * time.Millisecond)
+		// add a small amount of jitter to the timestamps so they don't have perfect amount of gap
+		jitter := time.Duration(rand.Intn(100)) * time.Millisecond
+		// update since so the next iteration goes up in time
+		since = since.Add(timeBetweenEachFlow + jitter)
+		// truncate the time if it goes above until
+		if since.After(until) {
+			since = until
+		}
+		t := since
```

Comment:
The clamping can lead to duplicated time here, e.g.

```
% ./cmd/fake flow --since 200ms --until 0 --count 10
now	2023-04-14 14:00:00 +0000 UTC
since	2023-04-14 13:59:59.8 +0000 UTC
until	2023-04-14 14:00:00 +0000 UTC

TIMESTAMP             SOURCE                 DESTINATION            TYPE                        VERDICT     SUMMARY
Apr 14 13:59:59.819   10.2.199.168           10.196.197.25          from-network                FORWARDED
Apr 14 13:59:59.911   10.20.29.42            10.233.248.150         DBG_CAPTURE_POINT_UNKNOWN   FORWARDED
Apr 14 13:59:59.957   10.20.29.42:13535      10.233.248.150:12577   DBG_CAPTURE_POINT_UNKNOWN   FORWARDED
Apr 14 13:59:59.973   10.105.149.59          10.114.130.232         UNKNOWN                     FORWARDED
Apr 14 13:59:59.997   10.202.102.254         10.198.180.143         UNKNOWN                     FORWARDED
Apr 14 14:00:00.000   10.202.102.254:7085    10.198.180.143:19125   12                          FORWARDED
Apr 14 14:00:00.000   10.224.63.76           10.165.72.174          UNKNOWN                     FORWARDED
Apr 14 14:00:00.000   10.20.29.42            10.233.248.150         to-overlay                  FORWARDED
Apr 14 14:00:00.000   10.126.182.242:16355   10.166.183.33:42835    DBG_CAPTURE_POINT_UNKNOWN   FORWARDED
Apr 14 14:00:00.000   10.20.29.42:34384      10.233.248.150:12667   to-network                  FORWARDED
```

I think we should evenly distribute time across the full requested range:

```suggestion
		fullRange := until.Sub(since)
		if int64(fullRange) < int64(opts.count) {
			return errors.New("time range it too short for the requested count")
		}
		winRange := fullRange / time.Duration(opts.count)
		jitter := rand.Int63n(int64(winRange))
		t := since.Add(time.Duration(i)*winRange + time.Duration(jitter))
```

Here `fullRange` and `winRange` should replace `timeRangeDuration` and `timeBetweenEachFlow` and declared outside the loop. Just an example suggestion.

Reply:
Without the clamping, we can produce values that go beyond the time range due to the jitter. I can't figure out a better way to do it though. I guess I need to make jitter smaller as we get closer to the end.

Reply:
Ah but you scale jitter based on the window, so it works. I see.